Merge develop into salling-service

JJTM20 opened the pull request.

sebxGit committed:
Merge remote-tracking branch 'origin/frontend-favouriterecipeintegrationwithredux' into frontend-savePreviousRecipes

sebxGit committed:
added feature to save previously hearted recipes and review them

sebxGit committed:
minor changes to pr

sebxGit committed:
added to log ind instead of log in

Casper0913 committed:
Addes fetch using rtk query for login and create user

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Casper0913 committed:
Fixes runninng error

Casper0913 committed:
Updates to Generate Recipes Button Handler

sebxGit committed:
added radius slider

sebxGit committed:
Update RadiusSlider.jsx

sebxGit committed:
Merge branch 'develop' into frontend-savePreviousRecipes

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

sebxGit committed:
added buttons

sebxGit committed:
Merge branch 'develop' into frontend-radiusslider

Casper0913 committed:
Fixes paths for userApiSlice

sebxGit committed:
now requires geolocation to use radius functionality

sebxGit committed:
undid change

Casper0913 committed:
Small fixes for test post

Lildhansen committed:
fixed warnings + fjernet hjertede(kunne ikke lige fixe det med at man kunne scrolle bag modalen)

Casper0913 committed:
Save user token and username inside an cookie

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Casper0913 committed:
Fixes split error in cookie

sebxGit committed:
Merge branch 'develop' into frontend-savePreviousRecipes

sebxGit committed:
added tests and fixed modal bug

sebxGit committed:
Merge branch 'frontend-savePreviousRecipes' into frontend-radiusslider

sebxGit committed:
added classes for toggleButton and changed to better namings

sebxGit committed:
added tests

Casper0913 committed:
Add logout communication between frontend and backend

Casper0913 committed:
fixes cookies for creating user + fixes response

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Casper0913 committed:
code clean up + added function to retrive cookie

Lildhansen committed:
made it so exlcude ingredient and owned ingredients cant have the same values. also added tests for addIngredientsInput

Casper0913 committed:
Delete addCookie.js

Casper0913 committed:
Updates to userSlice + code cleanup

Casper0913 committed:
Fixes rename of toggleswitch

sebxGit committed:
changed styling, now saves radius if switching to mulige butikker and back

Lildhansen committed:
it works on my machine

sebxGit committed:
bugged menu but remove works.

sebxGit committed:
testing now works.

sebxGit committed:
Merge branch 'develop' into frontend-radiusslider

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Lildhansen committed:
finished the addIngredientsFormTest

Casper0913 committed:
Moved user endpoints in to own file

Casper0913 committed:
Fix for error when logging out

Casper0913 committed:
Updated userSlice + fixes warnings + code clean up

Casper0913 committed:
Added comments to code + small updates to userSlice

Casper0913 committed:
Encodes input for request using encodeURIComponent

done to avoid request error for symbols

Casper0913 committed:
Code clean up for store #67

Lildhansen committed:
forstår ikke det double click men har fixed så man kan lukke favorites igen og formattet filerne (samt gjort favorites box lidt mere læsbar)

Lildhansen committed:
Merge branch 'develop' into frontend-savePreviousRecipes

Casper0913 committed:
type fixed + consistent name for loggedIn + small code cleanup

Lildhansen committed:
selecting favorite recipes is more simple now - and you can remove them (with a confirmation alert)

Lildhansen committed:
added tests for the favoriteBox (kan ikke teste mere i dybden end det her pga redux - kan fx ikke tjekke om elementer rent faktisk bliver slettet, fordi man skal logge ind som user først hvilket ikke virker)

Jaffenheimer committed:
Added docstrings form most of the code, e.g where they are necessary. Also i have made small changes and removed unused code. Will refactor OpenAiService.cs tmr.

Jaffenheimer committed:
Formatted the code and optimized usings

Jaffenheimer committed:
Added docstrings for most of the controller the one missing OpenAI needs refactoring

Jaffenheimer committed:
Refactored offer endpoint and OpenAiService.cs

Jaffenheimer committed:
Small change

Lildhansen committed:
removed console log

Lildhansen committed:
no error for test and danish instead of english test (also some formatting it seems)

Lildhansen committed:
more clean solution

Jaffenheimer committed:
Added recipe history aswell as endpoints to facilitate this, including a recipe endpoint to use when logged in

Jaffenheimer committed:
Changed new table to be embedded recourse

Lildhansen committed:
Merge pull request #154 from Jaffenheimer/frontend-savePreviousRecipes

Frontend save previous recipes

Jaffenheimer committed:
Added amount to RecipeRequest.cs

Jaffenheimer committed:
Fixed typo

Jaffenheimer committed:
Removed endpoint that should not be there

sebxGit committed:
Merge branch 'develop' into frontend-radiusslider

Jaffenheimer committed:
Merge pull request #173 from Jaffenheimer/Docstrings-for-all-functions

Added docstrings to all of backend, refactored OpenAiService and added recipe history

sebxGit committed:
Merge pull request #160 from Jaffenheimer/frontend-radiusslider

Frontend radiusslider

Jaffenheimer committed:
Update UserSessionRepository.cs

Made change to account for utc

NikolajKrogh committed:
Fixes #165

Jaffenheimer committed:
Added deletion of deprecated tokens

NikolajKrogh committed:
Merge branch 'develop' into align-cross-with-field

NikolajKrogh committed:
Merge pull request #178 from Jaffenheimer/align-cross-with-field

Fixes #165

Lildhansen committed:
remove all button fix

Jaffenheimer committed:
Added ( because i am dumbo

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Casper0913 committed:
Merge branch 'develop-connect_frontend_to_backend' of https://github.com/Jaffenheimer/p7-internet into develop-connect_frontend_to_backend

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Casper0913 committed:
Fixes comments to pr

Casper0913 committed:
Added test for cookie

Casper0913 committed:
Formating + Changes comments to english

Lildhansen committed:
fixet så den kun tjekker om du har valgt butikker hvis du er under "vælg butikker". har ændret på storeslice så den dækker mere end bare stores

Casper0913 committed:
Ændre tekst i toast

Casper0913 committed:
Update LoginBox.jsx

Casper0913 committed:
Fixes serializable warning when testing

Lildhansen committed:
split the toggle test into toggleButtonTest, radiusSliderTest, and StoreSelectionTest (+ added more tests)

Lildhansen committed:
cleanned up imports

NikolajKrogh reviewed (approved):
Changes er fra dev, så tænker det er okay

Jaffenheimer committed:
Added support for ingredients in the DB we can now check a string coming from the frontend, to see if it matches an ingredient in the db

Lildhansen committed:
radius and stores now keep their value when changing between them

Jaffenheimer committed:
Update README.md

Lildhansen committed:
more cleanup

Casper0913 committed:
Moved input validation functions into a file

Lildhansen committed:
Merge branch 'develop' into ingredientNonDuplicate

Lildhansen committed:
det var da irriterende at jeg hele tiden skal rette det her

Lildhansen committed:
Merge branch 'ingredientNonDuplicate' of https://github.com/Jaffenheimer/p7-internet into ingredientNonDuplicate

Lildhansen committed:
Merge pull request #171 from Jaffenheimer/ingredientNonDuplicate

Ingredients added to exclude list cannot be identical to those added to owned list (and vice versa)

Lildhansen committed:
updated such that recipeSelectionContainerRIght now have the tests with the logic of the togglebutton and radiusslider

Casper0913 committed:
Combined Create and Login user + add cookie functions + codeclean  up  + more

- Split Create user and login user into one function
- Added function to add and delete cookies
- Moved getLocation into file
- Moved inputValidation into file
- Code Cleanup

Casper0913 committed:
Update ProfilePicture.jsx

Casper0913 committed:
Merge branch 'develop' into develop-connect_frontend_to_backend

Casper0913 committed:
Added test for cookiehandler

Casper0913 committed:
Removed axios from project + addeding loading state to login/create user function

- Also removed example file from project

Casper0913 committed:
prettier files worked on + code cleanup

Casper0913 committed:
added printing error when creating and login in

Casper0913 committed:
Merge pull request #172 from Jaffenheimer/develop-connect_frontend_to_backend

Develop connect frontend to backend (Create User, Login User, Logout user)

Lildhansen committed:
fixed tests?

Lildhansen committed:
Merge branch 'develop' into selectShopsFix

Lildhansen committed:
Merge pull request #194 from Jaffenheimer/selectShopsFix

fixet så den kun tjekker om du har valgt butikker hvis du er under "vælg butikker". har ændret på storeslice så den dækker mere end bare stores (offerSlice nu)

Jaffenheimer committed:
Small changes, and fix to wrong error message

Lildhansen committed:
hot reload er fixed nu

Lildhansen committed:
Merge pull request #196 from Jaffenheimer/fixHotReload

hot reload er fixed nu

Jaffenheimer committed:
Merge branch 'develop' into ingredient-db

Jaffenheimer committed:
Merge pull request #187 from Jaffenheimer/ingredient-db

Added support for ingredients in the DB we can now check a string com…
